pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -498,6 +540,182 @@ var runCancelled = PipelineRunState{{
 	Run:          withRunCancelled(newRun(runs[0])),
 }}
 
+var noneStartedStateMatrix = PipelineRunState{{
+	PipelineTask: &pts[15],
+	TaskRunNames: []string{"pipelinerun-mytask1"},
+	TaskRuns:     nil,
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}, {
+	PipelineTask: &pts[16],
+	TaskRunNames: []string{"pipelinerun-mytask3"},
+	TaskRuns:     nil,
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}}
+
+var oneStartedStateMatrix = PipelineRunState{{
+	PipelineTask: &pts[15],
+	TaskRunNames: []string{"pipelinerun-mytask1"},
+	TaskRuns:     []*v1beta1.TaskRun{makeStarted(trs[0])},
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}, {
+	PipelineTask: &pts[16],
+	TaskRunNames: []string{"pipelinerun-mytask2"},
+	TaskRuns:     nil,
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}}
+
+var oneFinishedStateMatrix = PipelineRunState{{
+	PipelineTask: &pts[15],
+	TaskRunNames: []string{"pipelinerun-mytask1"},
+	TaskRuns:     []*v1beta1.TaskRun{makeSucceeded(trs[0])},
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}, {
+	PipelineTask: &pts[16],
+	TaskRunNames: []string{"pipelinerun-mytask2"},
+	TaskRuns:     nil,
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}}
+
+var oneFailedStateMatrix = PipelineRunState{{
+	PipelineTask: &pts[15],
+	TaskRunNames: []string{"pipelinerun-mytask1"},
+	TaskRuns:     []*v1beta1.TaskRun{makeFailed(trs[0])},
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}, {
+	PipelineTask: &pts[16],
+	TaskRunNames: []string{"pipelinerun-mytask2"},
+	TaskRun:      nil,
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}}
+
+var finalScheduledStateMatrix = PipelineRunState{{
+	PipelineTask: &pts[15],
+	TaskRunNames: []string{"pipelinerun-mytask1"},
+	TaskRuns:     []*v1beta1.TaskRun{makeSucceeded(trs[0])},
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}, {
+	PipelineTask: &pts[16],
+	TaskRunNames: []string{"pipelinerun-mytask2"},
+	TaskRuns:     []*v1beta1.TaskRun{makeScheduled(trs[1])},
+	ResolvedTaskResources: &resources.ResolvedTaskResources{
+		TaskSpec: &task.Spec,
+	},
+}}
+
+var retryableFinalStateMatrix = PipelineRunState{{
```

Comment:
is this used anywhere? are there test cases covering remaining retries?

Reply:
good catch - removed this unused variable 

yes, the test cases for retries are defined with the tests for retries scenarios - see `TestGetNextTasks`